dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataSourceService.java

```diff
@@ -526,8 +524,7 @@ public String buildParameter(String name, String desc, DbType type, String host,
             parameterMap.put(Constants.PRINCIPAL,principal);
         }
         if (other != null && !"".equals(other)) {
-            LinkedHashMap<String, String> map = JSON.parseObject(other, new TypeReference<LinkedHashMap<String, String>>() {
-            });
+            LinkedHashMap<String, String> map = JSONUtils.parseObject(other, LinkedHashMap.class);
```

Comment:
suggest use TypeReference to parse 'other' variable
`TypeReference<HashMap<String, String>> typeRef 
  = new TypeReference<HashMap<String, String>>() {};
Map<String, String> map = mapper.readValue(jsonInput, typeRef);`